### PR TITLE
Use openseadragon for image previews

### LINKED ISSUE #1564: Provide higher resolution images and allow zoom for images that are associated with a media object
### Expected behavior
The desired outcome is to present higher resolution images and allow the user to zoom in if needed. 

### Actual behavior
Images of the original housing of archival media, along with disc labels, cassette inserts, album covers, and liner notes are typically included as part of a media object. The purpose of providing these images is to allow users to view the writing, notes, and drawings that are considered part of the archival media object. In the current media player, users are unable to see the images clearly and are unable to zoom in to read the small text, which impedes researcher and scholarship, and presents accessibility issues.

It appears that across all access restrictions, a low resolution image file is provided in the media player. 

### Steps to reproduce the behavior
Examples of this issue: https://purl.stanford.edu/bm996pp6037, https://purl.stanford.edu/cz768mr8354, https://purl.stanford.edu/df142py2033.

### PR DESCRIPTION
**Pull request by jcoyne**
fixes #1564

### CODE REVIEW

File: app/components/embed/media_preview_image_component.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Embed
+  # Displays an image (e.g. photo of the media) for a media resource
+  class MediaPreviewImageComponent < ViewComponent::Base
+    include StacksImage
+
+    # @param [String] druid the object identifier
+    # @param [ResourceFile] file the file to display
+    # @param [Integer] file_index the offset of this file in the resource
```
Comment (mjgiarlo) [inline app/components/embed/media_preview_image_component.rb:10]: `s/file_index/resource_index/` ?

Comment (mjgiarlo) [reply app/components/embed/media_preview_image_component.rb:10]: And not sure if the rest of the verbiage also needs changing.

File: config/initializers/assets.rb [hunk 2]
```diff
@@ -6,9 +6,9 @@
...
 # Add Yarn node_modules folder to the asset load path.
-# Rails.application.config.assets.paths << Rails.root.join('node_modules')
+Rails.application.config.assets.paths << Rails.root.join('node_modules/openseadragon/build/openseadragon/images')
 
 # Precompile additional assets.
 # application.js, application.css, and all non-JS/CSS in the app/assets
 # folder are already added.
-# Rails.application.config.assets.precompile += %w( admin.js admin.css )
\ No newline at end of file
+# Rails.application.config.assets.precompile += Rails.root.join('vendor/assets/images').glob('*.png').map(&:to_s)
```
Comment (mjgiarlo) [inline config/initializers/assets.rb:14]: Intended change or cruft?

File: app/components/embed/media_preview_image_component.rb (hunk 1 above)
Comment (jcoyne) [reply app/components/embed/media_preview_image_component.rb]: Thanks. Fixed that up.

File: config/initializers/assets.rb (hunk 2 above)
Comment (jcoyne) [reply config/initializers/assets.rb]: Cruft. I've removed it.

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**